### PR TITLE
add diamonds given view and diamonded posts page

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

File: src/app/creator-profile-page/creator-diamonds/creator-diamonds.component.ts [hunk 1]
```diff
@@ -35,12 +51,42 @@ export class CreatorDiamondsComponent implements OnInit {
...
   counter(num: number) {
     return Array(num);
   }
 
-  usernameTruncationLength(): number {
-    return this.globalVars.isMobile() ? 10 : 12;
+  getNoDiamondsMessage() {
+    return this.showDiamondsGiven
+      ? `@${this.profile.Username} has not given any diamonds yet.`
+      : `No has sent @${this.profile.Username} a diamond yet.`;
```
Comment (maebeam) [inline src/app/creator-profile-page/creator-diamonds/creator-diamonds.component.ts:61]: `@${this.profile.Username} has not received any diamonds yet.`

File: src/app/feed/feed-post/feed-post.component.ts [hunk 2]
```diff
@@ -80,6 +80,8 @@ export class FeedPostComponent implements OnInit {
   @Input() showQuotedContent = true;
   @Input() hoverable = true;
 
+  @Input() showDiamondsFromSender = false;
```
Comment (maebeam) [inline src/app/feed/feed-post/feed-post.component.ts:83]: rename this to something like `showReplyingTo`?

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    "-1": new Promise((resolve) => {
+      resolve([]);
+    }),
+  };
+
+  lastPage = null;
+
+  static PAGE_SIZE = 10;
+
+  getDatasource() {
```
Comment (maebeam) [inline src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:53]: we should really make this infinite scroller component reusable. don't let this block merge though

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        if (!this.receiverProfileEntryResponse) {
+          this.receiverProfileEntryResponse = res.ReceiverProfileEntryResponse;
+        }
+        if (!this.senderProfileEntryResponse) {
+          this.senderProfileEntryResponse = res.SenderProfileEntryResponse;
+        }
+        posts.map((post) => (post.ProfileEntryResponse = res.ReceiverProfileEntryResponse));
+
+        let lastDiamondLevel = this.lastDiamondLevelOnPage[page - 1];
+        console.log(lastDiamondLevel);
```
Comment (maebeam) [inline src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:129]: remove

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        for (let ii = 0; ii < posts.length; ii++) {
+          posts[ii].ProfileEntryResponse = this.receiverProfileEntryResponse;
+          if (posts[ii].DiamondsFromSender != lastDiamondLevel) {
+            posts[ii].ShowDiamondDivider = true;
+            lastDiamondLevel = posts[ii].DiamondsFromSender;
+          }
+        }
+        console.log(posts.length);
```
Comment (maebeam) [inline src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:137]: remove

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+          this.senderProfileEntryResponse = res.SenderProfileEntryResponse;
+        }
+        posts.map((post) => (post.ProfileEntryResponse = res.ReceiverProfileEntryResponse));
+
+        let lastDiamondLevel = this.lastDiamondLevelOnPage[page - 1];
+        console.log(lastDiamondLevel);
+        for (let ii = 0; ii < posts.length; ii++) {
+          posts[ii].ProfileEntryResponse = this.receiverProfileEntryResponse;
+          if (posts[ii].DiamondsFromSender != lastDiamondLevel) {
+            posts[ii].ShowDiamondDivider = true;
```
Comment (maebeam) [inline src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:133]: we shouldn't be editing the PostEntryResponse to include rendering data. it should only contain the response from the server.

make a new type called something like DiamondsPost that contains a PostEntryResponse, ShowDiamondDivider, and DiamondsFromSender please.

**Review by maebeam — COMMENTED** [review]
minor fixes pls

File: src/app/creator-profile-page/creator-diamonds/creator-diamonds.component.ts (hunk 1 above)
Comment (lazynina) [reply src/app/creator-profile-page/creator-diamonds/creator-diamonds.component.ts:61]: :facepalm: thanks!

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts (hunk 3 above)
Comment (lazynina) [reply src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:53]: agreed it's on my list.

File: src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts (hunk 6 above)
Comment (lazynina) [reply src/app/diamond-posts-page/diamond-posts/diamond-posts.component.ts:133]: core's PostEntryResponse struct includes `DiamondsFromSender`, but I'll create a new type for DiamondPost that is a post and the show diamond divider attribute

**Pull request merged**